telematic_system/telematic_cloud_messaging/src/main/java/com/telematic/telematic_cloud_messaging/message_converters/JSON2KeyValuePairsConverter.java

```diff
@@ -62,10 +70,15 @@ else if (value.toString().toLowerCase().strip().equals("nan")){
                     }                    
                 }              
                 
-                if (json.keySet().size() != key_count) {
+                if (!is_ignored && json.keySet().size() != key_count) {
                     pairs += ",";
                 }
-            }
+                //If last field is ignored, remove the tail comma
+                if(is_ignored && json.keySet().size() == key_count && pairs.length() >= 1)
```

Comment:
Shouldn't we also be reducing the key count is a field is ignored?

Reply:
If the key count is reduced we cannot tell if we reach the end of this JSON.